packages/cli/README.md

```diff
@@ -194,6 +134,67 @@ While the usage of ENV variables is minimized, CLI still uses WNS related variab
 
 ENV variables are also used to pass configuration between CLI and spawned processes, but this happens transparently for CLI user.
 
+## Upgrade
+
+An older version of the CLI could be upgraded via `dx upgrade` command.
+
+```
+$ dx version
+v1.0.1-beta.15
+
+$ dx upgrade --force
+Found extensions: @dxos/cli-data, @dxos/cli-signal, @dxos/cli-bot, @dxos/cli-app
+✔ Uninstalling @dxos/cli-data
+✔ Uninstalling @dxos/cli-signal
+✔ Uninstalling @dxos/cli-bot
+✔ Uninstalling @dxos/cli-app
+✔ Uninstalling @dxos/cli
+✔ Installing @dxos/cli
+✔ Installing @dxos/cli-app
+✔ Installing @dxos/cli-bot
+✔ Installing @dxos/cli-signal
+✔ Installing @dxos/cli-data
+
+$ dx version
+v1.0.1-beta.16
+```
+
+## Troubleshooting
+
+Most of the weirdness in CLI behaviour could be caused by any combination of previously installed CLIs and extensions, especially from the beta channel;
+
+Check latest available version:
+
+```bash
+$ yarn info @dxos/cli versions --json | jq '.data[-1]'
+```
+
+Check installed version:
+
+```bash
+$ dx version
+```
+
+If those outputs are different, make sure to remove old versions of `wire`.
+Remove old CLI and extensions, installed globally.
+
+Starting v1.0.0-beta.30, `dx uninstall` and `dx upgrade` commands are available.
+
+To remove CLI and all extensions:
+
+```
+$ dx uninstall --npm-client yarn
+```
+
+To force upgrade CLI and all installed extensions to the latest:
+
+```
+$ dx upgrade --npm-client yarn --force
+```
+
+`--version` attribute could be supplied in order to upgrade/downgrade to a specific version.
+
+
 ## Commands
```

Comment:
@Shro3 I guess it would be good to move the "Commands" section right after "Installation".

Thinking about how a common user will use this, would be after installation, it would try to throw some commands. So I think it makes sense to have it there, and leave the "Troubleshooting" section at the end of the file. It's usually the place where everyone leaves it.